Increments for iP

Shuggan opened the pull request:
# Duke
> "My laptop before me, and the world at my fingertips. Take pride coder" - Coder's creed [source](https://drive.google.com/file/d/1fsPKB-ANbXKy7eP5qOAudEwgTO2t4PNA/view?usp=sharing)
### Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use
- Full of cute profile images

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 hehe!

And it is FREE! Wow!

Features:

- [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
no message

Commit:
Add "delete" feature to remove a task from the list

Commit:
Add A-TextUiTesting to Duke environment

joellow88 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                    }
+                    Task t = new Task.Todo(textInput.substring(5));
+                    taskList.add(t);
+                    String output = String.format("Got it. I've added this task:\n%s\nNow you have %d tasks in the list", t.toString(), taskList.size());
```
This output line seems very long, is it possible to split them into 2 lines?
A similar issue appears in line 32, 78, 88.

joellow88 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+            if (textInput.equalsIgnoreCase("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < taskList.size(); i++) {
+                    System.out.println(i + 1 + ". " + taskList.get(i).toString());
```
This manual string formatting works, however, have you considered using Java's builtin String.format method?

joellow88 commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception{
```
For this class and your Task class, can you indent it with 4 spaces instead of 2?

joellow88 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                taskList.remove(i - 1);
+                String output = String.format("Got it. I've removed this task:\n%s\nNow you have %d tasks in the list", t.toString(), taskList.size());
+                System.out.println(output);
+                continue;
```
I find the use of multiple `continue` instructions in the code strange. The while loop already has a looping behaviour once it reaches the end of the loop, so using `continue` to return to the start of the loop should not be a default behaviour. What is your rationale for designing it this way?

Rather, I would suggest the following structure:
```java
boolean isRunning = true;
while (isRunning) {
    if (textInput.equalsIgnoreCase("bye")) {
        isRunning = false;
    } else if (stmt) {
        code;
    } else {
        code;
    }
}
```

Alternatively, you may consider parsing the command first, then using a switch-case statement to handle various cases.

joellow88 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+import java.util.ArrayList;
```
Would prefer if the code was in a package, eg: package duke.

joellow88 reviewed (commented):
Up to Week 2 (Level-6), the code is functional and meets all the weekly requirements. There are minor issues regarding the coding standard, and a few suggestions for refactoring in the Duke.java class.

Commit:
Changes until MoreOOP

Commit:
no message

Commit:
Added "dates" feature

Commit:
Created package for "duke"

Commit:
Moved files to package folder

Commit:
ADD gradle functionality

Commit:
ADD JUnit tests

Commit:
ADD find feature

Commit:
ADD Javadocs to public methods and classes

Commit:
IMPROVE code quality

Commit:
ADD Varargs where applicable

Commit:
no message

Commit:
ADD GUI feature

Commit:
Add assertions to ensure that the user inputs are always valid.

These are mainly done when the user is adding tasks to the TaskList through "deadline" and "event".

Adding these assertions improves Duke's user experience and handles invalid user inputs better.

Commit:
Improve code quality with better abstraction, closing platform features, and CSS styling.

There are several ways code quality was improved. For example, better OOP was implemented by abstracting away code from initialize() to setDuke() in MainWindow.java.

Improving the code quality helps to improve readability and adding CSS styling helps to improve the user experience.

Commit:
Add postpone feature to change the dates of deadlines to a later date.

This can be done with the following structure: "postpone <task name> /to <new deadline>".

Adding this postpone feature can improve the functionalities available to the user.

Shuggan committed:
Update README.md

Added user guide for better readibility.

Shuggan committed:
Add a representative screenshot

Shuggan committed:
Merge pull request #2 from Shuggan/branch-A-Assertions

Add assertions to ensure that the user inputs are always valid.

Shuggan committed:
Merge pull request #3 from Shuggan/branch-A-CodeQuality

Branch a code quality

Shuggan committed:
Update README.md
